app/modules/ads/bill-the-lizard-wrapper.js

```diff
@@ -205,12 +205,14 @@ export const billTheLizardWrapper = {
     const { context } = window.Wikia.adEngine;
     const { billTheLizard } = window.Wikia.adServices;
 
-    context.set('services.billTheLizard.parameters.cheshirecat', {
-      bids: serializeBids(bidPosKeyVal),
-    });
-    cheshirecatCalled = true;
+    serializeBids(bidPosKeyVal).then(bids => {
+      context.set('services.billTheLizard.parameters.cheshirecat', {
+        bids,
+      });
+      cheshirecatCalled = true;
```

Comment:
`cheshirecatCalled` was desired to prevent duplicated cheshirecat call in line 155. Are you 100% sure that this promise will be resolved before `renderEnded` on slot?

Reply:
I don't know, good catch. I will have to inspect it in more detail.

Reply:
Let's check that in data QA.

Reply:
Why not chain promises: serializeBids -> on render end ?

Reply:
Could you provide an example?